VideoRenderer/VideoRenderer/SystemPlayerObserver.swift

```diff
@@ -163,13 +165,13 @@ public final class SystemPlayerObserver: NSObject {
             emit(.didChangeUrl(from: oldUrl, to: newUrl))
             
         case #keyPath(AVPlayerItem.status):
-            let oldStatus = oldValue().flatMap(AVPlayerItemStatus.init)
-            guard let newStatus = newValue().flatMap(AVPlayerItemStatus.init) else {
-                fatalError("Unexpected nil in AVPlayerItem.status value!")
+            guard let newStatus = newValue().flatMap(AVPlayerItemStatus.init) else { fatalError("Unexpected nil in AVPlayerItem.status value!") }
+            switch newStatus {
+            case .unknown: emit(.didChangeItemStatusToUnknown())
+            case .readyToPlay: emit(.didChangeItemStatusToReadyToPlay())
+            case .failed: return emit(.didChangeItemStatusToFailed(error: player.error ?? player.currentItem?.error))
```

Comment:
Is `error` of the `player` observable?
Maybe we can observe it and instead send failed when we received an error?

Reply:
In this case we should observe `player` and `currentItem` `error`. Is it ok?

Reply:
Let's leave only `currentItem.error`